### PR TITLE
Sans gui beam center finder

### LINKED ISSUE #20822: Implement Beam centre finder to work with SANS STATE
The new sans Gui does not currently have the capability to find the beam center.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
Description of work.
This work implements the BeamCentreFinder command for ISIS sans using the new backend. 
**To test:**
To test run the following two scripts in mantid and check that they produce the same results. 

from sans.command_interface.ISISCommandInterface import *
Clean()
MASKFILE = FileFinder.getFullPath('MaskSANS2DReductionGUI.txt')
BATCHFILE = FileFinder.getFullPath('sans2d_reduction_gui_batch.csv')
UseCompatibilityMode()
SANS2D()
MaskFile(r'C:\Users\tgz88916\Documents\MantidBuild\ExternalData\Testing\Data\SystemTest\SANS2D\MaskSANS2DReductionGUI.txt')
AssignSample('22048')
AssignCan('22023')
TransmissionSample('22041', '22024')
TransmissionCan('22024', '22024')
centre = FindBeamCentre(rlow=41.0, rupp=280.0, xstart=float(150)/1000., ystart=float(-160)/1000., tolerance=0.0001251, MaxIter=3, reduction_method=True)

and

from ISISCommandInterface import *
Clean()
MASKFILE = FileFinder.getFullPath('MaskSANS2DReductionGUI.txt')
BATCHFILE = FileFinder.getFullPath('sans2d_reduction_gui_batch.csv')
SANS2D()
MaskFile(r'C:\Users\tgz88916\Documents\MantidBuild\ExternalData\Testing\Data\SystemTest\SANS2D\MaskSANS2DReductionGUI.txt')
AssignSample('22048')
AssignCan('22023')
TransmissionSample('22041', '22024')
TransmissionCan('22024', '22024')
centre = FindBeamCentre(rlow=41.0, rupp=280.0, xstart=float(150)/1000., ystart=float(-160)/1000., tolerance=0.0001251, MaxIter=3)

The results are shown in the logging window. IN particular check that all the numbers after 3 iterations are the same. 
Fixes #20822 . 

**Release Notes** 
Have added to release notes

Also need to update here https://www.mantidproject.org/Scripting_SANS_Reductions_With_The_New_Reduction_Backend
once merged.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/sans/command_interface/ISISCommandInterface.py [hunk 1]
```diff
@@ -139,9 +139,9 @@ def createColetteScript(inputdata, format, reduced, centreit, plotresults, csvfi
     raise NotImplementedError("The creatColleteScript command is not implemented in SANS v2.")
 
 
-def FindBeamCentre(rlow, rupp, MaxIter=10, xstart=None, ystart=None, tolerance=1.251e-4,  find_direction=None):
-    _, _, _, _, _, _, _ = rlow, rupp, MaxIter, xstart, ystart, tolerance, find_direction  # noqa
-    raise NotImplementedError("The FindBeamCentre command is not implemented in SANS v2.")
+# def FindBeamCentre(rlow, rupp, MaxIter=10, xstart=None, ystart=None, tolerance=1.251e-4,  find_direction=None):
+#     _, _, _, _, _, _, _ = rlow, rupp, MaxIter, xstart, ystart, tolerance, find_direction  # noqa
+#     raise NotImplementedError("The FindBeamCentre command is not implemented in SANS v2.")
```
Comment (gemmaguest) [inline scripts/SANS/sans/command_interface/ISISCommandInterface.py]: Should this commented-out code be removed?

**Review by gemmaguest — COMMENTED** [review]
The output looks (probably!) correct - there's a factor of 1000 difference on the Pos values which I assume is just a difference in units? And the Res values are truncated in the second script's output, but look like they are otherwise the same:

First script:
```Iteration 3  PosX 0.15  PosY -0.145  ResX=34435.5409556  ResY=430000.737967```

Second script:
```Itr 3: (150.0    ,   -145.0   )    SX=34435.   SY=430000```

File: scripts/SANS/sans/command_interface/ISISCommandInterface.py (hunk 1 above)
Comment (Mantid-Matthew) [reply scripts/SANS/sans/command_interface/ISISCommandInterface.py]: Yes

**Review by gemmaguest — APPROVED** [review]
This looks good now :+1:.. The results from both scripts are the same, apart from the truncating of values in the old scripts, which is a minor issue.

**Pull request merged**